### ISSUE TITLE
VLAN support on pimox7

### ISSUE THREAD

**Issue by bronkoo** [pimox/pimox7#44]

**Comment by Sysadminfromhell** [comment]
Did you try to activate it via GUI ?

**Comment by bronkoo** [comment]
Yes, at first. But it is grayed out in ->Network ->Create ->Linyx Vlan

**Comment by Sysadminfromhell** [comment]
Did you activate a Linux bridge first and activate it then or otherwise? Because I can activate it.

**Comment by bronkoo** [comment]
Ok, thanks.
Just stumbled over: Default Bridge 'vmbr0' works without any changes - 'VLAN aware' is unmarked - and simple configure the 
'VLAN Tag' in network device for my CT's. Like KISS...

Not an issue currently, but 'VLAN Tag" is still grayed out in ->Network ->Create ->Linux Vlan.

**Comment by Sysadminfromhell** [comment]
> Ok, thanks. Just stumbled over: Default Bridge 'vmbr0' works without any changes - 'VLAN aware' is unmarked - and simple configure the 'VLAN Tag' in network device for my CT's. Like KISS...
> 
> Not an issue currently, but 'VLAN Tag" is still grayed out in ->Network ->Create ->Linux Vlan.

afaik you need to create a linux bond before you can create a linux vlan. Read the manual, I guess its described like this.

**Comment by bronkoo** [comment]
> afaik you need to create a linux bond before you can create a linux vlan. Read the manual, I guess its described like this.

sorry in the hell ;-), but that doesn't make sense... RTFM for sure.

**Comment by salman79uk** [comment]
> Did you try to 

> Did you try to activate it via GUI ?

Hi. For me activating the vlan aware feature on the bridge in the gui and then rebooting causes me to lose access to the pi and requiring me to start again from scratch with a reinstall. Tried a few times.

I don't get this issue on my other non pi nodes full proxmox clusters.

Any ideas.

**Comment by aanon4** [comment]
The vlan aware feature is a kernel compile option which the standard pi kernel doesn't enable. You can try this custom kernel which does have it enabled (see https://github.com/pimox/pimox7/issues/28)

**Comment by MakiseKurisu** [comment]
Also you may be able to do what you want with `openvswitch-switch` package.

**Comment by AlwinAntreich** [comment]
Just to add our trickery to get vlan-aware working, might not apply to your case. We needed to set our nics in `promisc` mode. Sounds like a bug. :)

https://community.mellanox.com/s/question/0D51T00008ansfP/vlan-aware-linux-bridging-is-not-functional-on-connectx4lx-card-unless-manually-put-in-promiscuous-mode
